src/inputosmpbf.cpp

```diff
@@ -48,10 +48,10 @@ extern std::function<bool(span_t<relation_t>)> relation_handler;
 extern bool verbose;
 struct field_t
 {
-    uint32_t key; // https://developers.google.com/protocol-buffers/docs/encoding#structure
-    uint8_t* pointer;
-    uint64_t length;
-    uint64_t value_uint64;
+    uint32_t key {0}; // https://developers.google.com/protocol-buffers/docs/encoding#structure
+    uint8_t* pointer {nullptr};
+    uint64_t length {0};
```

Comment:
got me thinking, do we expect stuff to be larger than 4GB realistically? the idea of the PBF is to split stuff in multiple bits, so in practice fields are actually quite small

we could have a layout like so:
```
struct field_t
{
    uint32_t key {0}; // https://developers.google.com/protocol-buffers/docs/encoding#structure
    uint32_t length {0};
    uint8_t* pointer {nullptr};
    uint64_t value_uint64 {0};
};

Reply:
we don't expect blocks >4G (not even close), but the `field_t` is designed to accept generic PBF encoded values. so the 64bit varint would be needed for decoding node/way/relation `id`s for example ...

Reply:
yes, indeed, forgot that